src/Data/List/Relation/Binary/Permutation/Propositional/Properties.agda

```diff
@@ -373,3 +374,28 @@ product-↭ (swap {xs} {ys} x y r) = begin
   (y * x) * product ys ≡⟨ *-assoc y x (product ys) ⟩
   y * (x * product ys) ∎
   where open ≡-Reasoning
+
+------------------------------------------------------------------------
+-- catMaybes
+
+catMaybes-↭ : xs ↭ ys → catMaybes xs ↭ catMaybes ys
+catMaybes-↭ refl = refl
+catMaybes-↭ (trans xs↭ ↭ys) = trans (catMaybes-↭ xs↭) (catMaybes-↭ ↭ys)
+catMaybes-↭ {xs = .(mx ∷ _)} {.(mx ∷ _)} (prep mx xs↭)
+  with mx
```

Comment:
`maybe` would be better than `with` here.

Reply:
See comment above.

Reply:
For reference, this is what you would need to write:
```agda
catMaybes-↭ {xs = .mx ∷ xs} {.mx ∷ ys} (prep mx xs↭)
  = maybe {B = λ mx → catMaybes (mx ∷ xs) ↭ catMaybes (mx ∷ ys)}
          (λ x → prep x $ catMaybes-↭ xs↭)
          (catMaybes-↭ xs↭)
          mx
```
Notice the necessity to introduce variables for the tails of the two lists.

Reply:
Right, that's worse.